src/lib.rs

```diff
@@ -972,7 +966,7 @@ fn parse_mail_recursive(
                 // if there is no terminating boundary, assume the part end is the end of the email
                 let ix_part_end =
                     find_from_u8_line_prefix(raw_data, ix_part_start, boundary.as_bytes())
-                        .unwrap_or_else(|| raw_data.len());
+                        .unwrap_or(raw_data.len());
```

Comment:
I'm surprised by this change, since it makes `len()` run more often. Is there documentation somewhere on this lint and why it's preferable?

Reply:
Thanks for the details! Given this was an explicit decision in clippy and not a bug the change seems fine to me.